js/src/client.ts

```diff
@@ -1358,34 +1361,91 @@ export class Client implements LangSmithTracingClientInterface {
     );
   }
 
+  private async _createNodeFetchBody(parts: MultipartPart[], boundary: string) {
+    // Create multipart form data manually using Blobs
+    const chunks: Blob[] = [];
+
+    for (const part of parts) {
+      // Add field boundary
+      chunks.push(new Blob([`--${boundary}\r\n`]));
+      chunks.push(
+        new Blob([
+          `Content-Disposition: form-data; name="${part.name}"\r\n`,
+          `Content-Type: ${part.payload.type}\r\n\r\n`,
+        ])
+      );
+      chunks.push(part.payload);
+      chunks.push(new Blob(["\r\n"]));
+    }
+
+    // Add final boundary
+    chunks.push(new Blob([`--${boundary}--\r\n`]));
+
+    // Combine all chunks into a single Blob
+    const body = new Blob(chunks);
+
+    // Convert Blob to ArrayBuffer for compatibility
+    const arrayBuffer = await body.arrayBuffer();
+    return arrayBuffer;
+  }
+
+  private async _createMultipartStream(
+    parts: MultipartPart[],
+    boundary: string
+  ) {
+    // Create a ReadableStream for streaming the multipart data
+    // Only do special handling if we're using node-fetch
+    const stream = new ReadableStream({
+      async start(controller) {
+        // Helper function to write a chunk to the stream
+        const writeChunk = async (chunk: string | Blob) => {
+          if (typeof chunk === "string") {
+            controller.enqueue(new TextEncoder().encode(chunk));
```

Comment:
Do we need to create a new TextEncoder every time?

Reply:
No